pkg/providers/instance/instance.go

```diff
@@ -333,7 +356,6 @@ func newVMObject(
 						},
 					},
 				},
-				CustomData: lo.ToPtr(launchTemplate.UserData),
```

Comment:
Rather than deleting here lets just do an inline ternary assignment 
```suggestion
                                   CustomData: lo.ToPtr(lo.Ternary(provisionMode == options.ProvisionModeBootstrappingClient, launchTemplate.AgentBakerCustomData, launchTemplate.SelfContainedCustomData))
```